pages/recording/[[...id]].tsx

```diff
@@ -161,13 +161,12 @@ function RecordingPage({
       setRecording(rec);
 
       const isTestReplay = rec.metadata?.test && rec.metadata?.source;
-
-      if (isTestReplay && !rec.workspace?.isTest) {
+      if (rec.private && isTestReplay && !rec.workspace?.isTest) {
         setExpectedError({
           content: "This recording is not available.",
           message: "The recording must belong to a test suite",
         });
-      } else if (!isTestReplay && rec.workspace?.isTest) {
+      } else if (rec.private && !isTestReplay && rec.workspace?.isTest) {
```

Comment:
nit: I find these conditionals a bit harder to read than (I think) they should be. Maybe we could tidy this up a bit while we're in there?
```js
      if (rec.private) {
        const isTestReplay = rec.metadata?.test != null && rec.metadata?.source != null;
        const isTestWorkspace = rec.workspace?.isTest == true;

        if (isTestReplay !== isTestWorkspace) {
          setExpectedError({
            content: "This recording is not available.",
            message: isTestReplay
              ? "The recording must belong to a test suite"
              : "The recording cannot be in a test suite",
          });
        }
      }
```

Reply:
Yep - i think that's an improvement